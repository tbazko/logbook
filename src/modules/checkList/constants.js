export const ADD_ITEM = 'stats/checkList/ADD_ITEM';
export const TOGGLE_ITEM_CHECKBOX = 'stats/checkList/TOGGLE_ITEM_CHECKBOX';
export const REMOVE_LIST_ITEM = 'stats/checkList/REMOVE_LIST_ITEM';
export const CHECK_ALL = 'stats/checkList/CHECK_ALL';
export const UNCHECK_ALL = 'stats/checkList/UNCHECK_ALL';
export const REMOVE_ALL = 'stats/checkList/REMOVE_ALL';
export const SET_DEFAULT_ITEMS = 'stats/checkList/SET_DEFAULT_ITEMS';
export const ADD_LOG_WITH_TIMESTAMP = 'stats/checkList/ADD_LOG_WITH_TIMESTAMP';
