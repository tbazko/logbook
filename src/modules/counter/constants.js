export const INCREMENT_COUNTER = 'app/counter/INCREMENT_COUNTER';
export const DECREMENT_COUNTER = 'app/counter/DECREMENT_COUNTER';
export const RESET_COUNTER = 'app/counter/RESET_COUNTER';
