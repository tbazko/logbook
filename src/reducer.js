import { combineReducers } from 'redux'
import * as reducers from 'reducers'

const rootReducer = combineReducers({
  activityTypes: reducers.activityTypes,
  activityCheckList: reducers.activityCheckList,
  historicalActivityTypes: reducers.historicalActivityTypes,
  validationError: reducers.validationError,
  activityLogs: reducers.activityLogs,
  activeLogTimestamp: reducers.activeLogTimestamp,
})

export default rootReducer
