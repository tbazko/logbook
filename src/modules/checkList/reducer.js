import _ from 'lodash';
import * as types from './constants'

const initialState = {
  items: null,
  logs: null,
}

// const initialState = {
//   items: {
//     [id]: { title, descr }
//   },
//   logs: {
//     [timestamp]: {
//       [id]: { completed }
//     }
//   }
// }

export default (state = initialState, action) => {
  const { type } = action;
  const logs = {};

  switch (type) {
    case types.ADD_ITEM:
      return {
        ...state,
        items: {
          ...state.items,
          [action.id]: { title: action.title },
        },
      }
    case types.TOGGLE_ITEM_CHECKBOX:
      return {
        ...state,
        logs: {
          ...state.logs,
          [action.timestamp]: {
            ..._.get(state, `logs[${action.timestamp}]`, undefined),
            [action.id]: {
              completed: !_.get(state, `logs[${action.timestamp}][${action.id}].completed`, true),
            },
          },
        },
      }
    case types.REMOVE_LIST_ITEM:
      return {
        ...state,
        items: _.omit(state.items, [`${action.id}`]),
      }
    // case types.CHECK_ALL:
    //   return state.map(item => ({ ...item, completed: true }))
    case types.SET_DEFAULT_ITEMS:
    case types.UNCHECK_ALL:
      if (state.logs && state.logs[action.timestamp]) return state

      if (!state.items) return { ...state, logs: null }
      Object.keys(state.items).forEach((item) => {
        logs[item] = { completed: false }
      });

      return {
        ...state,
        logs: {
          ...state.logs,
          [action.timestamp]: logs,
        },
      }
    // case types.REMOVE_ALL:
    //   return []

    default:
      return state
  }
};
