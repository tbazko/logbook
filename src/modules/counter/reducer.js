import * as types from './constants';


export const initialState = {
  value: 0,
};

export default (state = initialState, action) => {
  const { type } = action;

  switch (type) {
    case types.RESET_COUNTER:
      return {
        value: 0,
      };
    case types.DECREMENT_COUNTER:
      return {
        value: state.value - 1,
      };
    case types.INCREMENT_COUNTER:
      return {
        value: state.value + 1,
      };
    default:
      return state;
  }
};
