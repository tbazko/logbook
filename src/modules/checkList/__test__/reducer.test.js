import * as types from '../constants'
import reducer from '../reducer'

describe('CheckList Reducer', () => {
  let state;

  it('should return object with empty items and logs', () => {
    state = reducer(state, {
      type: 'TEST',
    });
    expect(state).toMatchSnapshot();
  });

  it('should add 3 objects to items', () => {
    state = reducer(state, {
      type: types.ADD_ITEM,
      id: '1',
      title: 'Test 1',
    });
    state = reducer(state, {
      type: types.ADD_ITEM,
      id: '2',
      title: 'Test 2',
    });
    state = reducer(state, {
      type: types.ADD_ITEM,
      id: '3',
      title: 'Test 3',
    });
    expect(state).toMatchSnapshot();
  });

  it('should save objects "2" and "3" in "logs" under "timestamp-1"', () => {
    state = reducer(state, {
      type: types.TOGGLE_ITEM_CHECKBOX,
      id: '2',
      timestamp: 'timestamp-1',
    });
    state = reducer(state, {
      type: types.TOGGLE_ITEM_CHECKBOX,
      id: '3',
      timestamp: 'timestamp-1',
    });
    expect(state).toMatchSnapshot();
  });

  it('should save object "2" in "logs" under timestamp "timestamp-2"', () => {
    state = reducer(state, {
      type: types.TOGGLE_ITEM_CHECKBOX,
      id: '2',
      timestamp: 'timestamp-2',
    });
    expect(state).toMatchSnapshot();
  });

  // it('should set completed for all elements to true', () => {
  //   state = reducer(state, {
  //     type: types.CHECK_ALL,
  //   });
  //   expect(state).toMatchSnapshot();
  // });

  // it('should set completed for all elements to false', () => {
  //   state = reducer(state, {
  //     type: types.UNCHECK_ALL,
  //   });
  //   expect(state).toMatchSnapshot();
  // });

  it('should set completed for all elements in "logs" under "timestamp-3" to "false"', () => {
    const emptyLogsState = reducer({ ...state, logs: null }, {
      type: types.SET_DEFAULT_ITEMS,
      timestamp: 'timestamp-3',
    });
    expect(emptyLogsState).toMatchSnapshot();
  });

  it('should remove object "1" from "items"', () => {
    state = reducer(state, {
      type: types.REMOVE_LIST_ITEM,
      id: '1',
    });
    expect(state).toMatchSnapshot();
  });

  it('should set activeCheckList to "active-timestamp"', () => {
    state = reducer(state, {
      type: types.SET_ACTIVE_CHECKLIST,
      timestamp: 'active-timestamp',
    });
    expect(state).toMatchSnapshot();
  });

  // it('should remove all elements and return empty array', () => {
  //   state = reducer(state, {
  //     type: types.REMOVE_ALL,
  //   });
  //   expect(state).toMatchSnapshot();
  // });
});
