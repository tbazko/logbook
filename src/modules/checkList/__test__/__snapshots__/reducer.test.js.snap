// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckList Reducer should add 3 objects to items 1`] = `
Object {
  "items": Object {
    "1": Object {
      "title": "Test 1",
    },
    "2": Object {
      "title": "Test 2",
    },
    "3": Object {
      "title": "Test 3",
    },
  },
  "logs": null,
}
`;

exports[`CheckList Reducer should remove object "1" from "items" 1`] = `
Object {
  "items": Object {
    "2": Object {
      "title": "Test 2",
    },
    "3": Object {
      "title": "Test 3",
    },
  },
  "logs": Object {
    "timestamp-1": Object {
      "2": Object {
        "completed": false,
      },
      "3": Object {
        "completed": false,
      },
    },
    "timestamp-2": Object {
      "2": Object {
        "completed": false,
      },
    },
  },
}
`;

exports[`CheckList Reducer should return object with empty items and logs 1`] = `
Object {
  "items": null,
  "logs": null,
}
`;

exports[`CheckList Reducer should save object "2" in "logs" under timestamp "timestamp-2" 1`] = `
Object {
  "items": Object {
    "1": Object {
      "title": "Test 1",
    },
    "2": Object {
      "title": "Test 2",
    },
    "3": Object {
      "title": "Test 3",
    },
  },
  "logs": Object {
    "timestamp-1": Object {
      "2": Object {
        "completed": false,
      },
      "3": Object {
        "completed": false,
      },
    },
    "timestamp-2": Object {
      "2": Object {
        "completed": false,
      },
    },
  },
}
`;

exports[`CheckList Reducer should save objects "2" and "3" in "logs" under "timestamp-1" 1`] = `
Object {
  "items": Object {
    "1": Object {
      "title": "Test 1",
    },
    "2": Object {
      "title": "Test 2",
    },
    "3": Object {
      "title": "Test 3",
    },
  },
  "logs": Object {
    "timestamp-1": Object {
      "2": Object {
        "completed": false,
      },
      "3": Object {
        "completed": false,
      },
    },
  },
}
`;

exports[`CheckList Reducer should set completed for all elements in "logs" under "timestamp-3" to "false" 1`] = `
Object {
  "items": Object {
    "1": Object {
      "title": "Test 1",
    },
    "2": Object {
      "title": "Test 2",
    },
    "3": Object {
      "title": "Test 3",
    },
  },
  "logs": Object {
    "timestamp-3": Object {
      "1": Object {
        "completed": false,
      },
      "2": Object {
        "completed": false,
      },
      "3": Object {
        "completed": false,
      },
    },
  },
}
`;
